pkgs/development/tools/build-managers/cmake/setup-hook.sh

```diff
@@ -49,7 +49,13 @@ cmakeConfigurePhase() {
 
     # Avoid cmake resetting the rpath of binaries, on make install
     # And build always Release, to ensure optimisation flags
-    cmakeFlags="-DCMAKE_BUILD_TYPE=Release -DCMAKE_SKIP_BUILD_RPATH=ON $cmakeFlags"
+    cmakeFlags="-DCMAKE_SKIP_BUILD_RPATH=ON $cmakeFlags"
+
+    if [ -n "$cmakeBuildType" ]; then
+        cmakeFlags="-DCMAKE_BUILD_TYPE=$cmakeBuildType $cmakeFlags"
```

Comment:
If one appends `-DCMAKE_BUILD_TYPE=Debug` to `cmakeFlags`, does that not override the setting made here? In which case, we need only the `dontAddBuildType` option.